ts/types/attachments/migrations.ts

```diff
@@ -142,29 +143,49 @@ export const loadData = async (attachment: any) => {
   return { ...attachment, data };
 };
 
+const handleDiskDeletion = async (path: string) => {
+  await deleteOnDisk(path);
+  try {
+    const exists = await pathExists(path);
+
+    // Note we want to confirm the path no longer exists
+    if (exists) {
+      throw Error('Error: File path still exists.');
+    }
+
+    window.log.debug(`deleteDataSuccessful: Deletion succeeded for attachment ${path}`);
+    return undefined;
+  } catch (err) {
+    window.log.warn(`deleteDataSuccessful: Deletion failed for attachment ${path} ${err}`);
+    return path;
+  }
+};
+
 //      deleteData :: (RelativePath -> IO Unit)
 //                    Attachment ->
 //                    IO Unit
-export const deleteData = () => {
-  return async (attachment: { path: string; thumbnail: any; screenshot: any }) => {
-    if (!isValid(attachment)) {
-      throw new TypeError('deleteData: attachment is not valid');
-    }
+export const deleteData = async (attachment: { path: string; thumbnail: any; screenshot: any }) => {
+  if (!isValid(attachment)) {
+    throw new TypeError('deleteData: attachment is not valid');
+  }
 
-    const { path, thumbnail, screenshot } = attachment;
-    if (isString(path)) {
-      await deleteOnDisk(path);
-    }
+  const { path, thumbnail, screenshot } = attachment;
 
-    if (thumbnail && isString(thumbnail.path)) {
-      await deleteOnDisk(thumbnail.path);
-    }
+  if (isString(path)) {
+    attachment.path = String(await handleDiskDeletion(path));
```

Comment:
Good catch. Fixed should be `path && isString(path)` right?

Reply:
is that what you meant?

Reply:
I see! Ended up having to change slightly because we can't assign attachment.path to undefined;

```
  if (path && isString(path)) {
    const pathAfterDelete = await handleDiskDeletion(path);
    if (isString(pathAfterDelete)) {
      attachment.path = pathAfterDelete;
    }
  }
```